code/modules/mining/mine_items.dm

```diff
@@ -249,3 +387,924 @@
+	w_class = 3.0
+
+/obj/item/weapon/rrf/examine(mob/user)
+	if(..(user, 0))
+		user << "It currently holds [stored_matter]/30 fabrication-units."
+
+/obj/item/weapon/rrf/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if (istype(W, /obj/item/weapon/rcd_ammo))
+
+		if ((stored_matter + 30) > 30)
+			user << "The RRF can't hold any more matter."
+			return
+
+		qdel(W)
+
+		stored_matter += 30
+		playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+		user << "The RRF now holds [stored_matter]/30 fabrication-units."
+		return
+
+	if (istype(W, /obj/item/weapon/rrf_ammo))
+
+		if ((stored_matter + 15) > 30)
+			user << "The RRF can't hold any more matter."
+			return
+
+		qdel(W)
+
+		stored_matter += 15
+		playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+		user << "The RRF now holds [stored_matter]/30 fabrication-units."
+		return
+
+/obj/item/weapon/rrf/afterattack(atom/A, mob/user as mob, proximity)
+
+	if(!proximity) return
+
+	if(istype(user,/mob/living/silicon/robot))
+		var/mob/living/silicon/robot/R = user
+		if(R.stat || !R.cell || R.cell.charge <= 0)
+			return
+	else
+		if(stored_matter <= 0)
+			return
+
+	if(!istype(A, /turf/simulated/floor))
+		return
+
+	if(locate(/obj/structure/track) in A)
+		return
+
+	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+	var/used_energy = 0
+
+	used_energy = 10
+
+	new /obj/structure/track(get_turf(A))
+
+	user << "Dispensing track..."
+
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.cell)
+			R.cell.use(used_energy)
+	else
+		stored_matter--
+		user << "The RRF now holds [stored_matter]/30 fabrication-units."
+
+
+/obj/structure/track
+	name = "mine track"
+	desc = "Just like your grandpappy used to lay 'em in 1862."
+	icon = 'icons/obj/smoothtrack.dmi'
+	icon_state = "track15"
+	density = 0
+	anchored = 1.0
+	w_class = 3
+	layer = 2.44
+
+/obj/structure/track/initialize()
+	..()
+	for(var/obj/structure/track/T in src.loc)
+		if(T != src)
+			qdel(T)
+	updateOverlays()
+	for (var/dir in cardinal)
+		var/obj/structure/track/R = locate(/obj/structure/track, get_step(src, dir))
+		if(R)
+			R.updateOverlays()
+
+/obj/structure/track/Destroy()
+	for (var/dir in cardinal)
+		var/obj/structure/track/R = locate(/obj/structure/track, get_step(src, dir))
+		if(R)
+			R.updateOverlays()
+	..()
+
+/obj/structure/track/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			qdel(src)
+			return
+		if(3.0)
+			return
+		else
+	return
+
+/obj/structure/track/attackby(obj/item/C as obj, mob/user as mob)
+
+	if (istype(C, /obj/item/stack/tile/floor))
+		var/turf/T = get_turf(src)
+		T.attackby(C, user)
+		return
+	if (istype(C, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = C
+		if(WT.remove_fuel(0, user))
+			user << "<span class='notice'>Slicing apart connectors ...</span>"
+		getFromPool(/obj/item/stack/rods, src.loc)
+		qdel(src)
+
+	return
+
+/obj/structure/track/proc/updateOverlays()
+	spawn(1)
```

Comment:
Why spawn(1)? Might as well just put `set waitfor = FALSE` at the top, unless you *need* the 1ds delay.